src/source_sender/mod.rs

```diff
@@ -48,17 +51,37 @@ impl Builder {
         }
     }
 
-    pub fn add_source_output(&mut self, output: SourceOutput) -> LimitedReceiver<EventArray> {
+    pub fn add_source_output(
+        &mut self,
+        output: SourceOutput,
+        component_key: ComponentKey,
+    ) -> LimitedReceiver<EventArray> {
         let lag_time = self.lag_time.clone();
+        let log_definition = output.schema_definition.clone().map(Arc::new);
```

Comment:
If it's being turned into an `Arc` anyways, would it make sense to move that `Arc` up into `SourceOutput`?